### PR TITLE
P4-2236 initial preparation work for improving the error handling/logging.  Contains some other refactorings as well.

### PR DESCRIPTION
**Pull request by matt-crowson-moj**
Changes include:

Updates to read.me file.
Renaming classes.
Adding exception config to improve error responses.
Adding missing application.yml which in turns exposes /health and /info to the base path.
Restricting the amount of data that can be queried to one months data to avoid trying to pull back too much data.
Removed dummy data from the cancellation tab in the JPC template spreadsheet.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/health/VersionOutputter.kt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import com.microsoft.applicationinsights.extensibility.ContextInitializer
+import org.slf4j.LoggerFactory
+import org.springframework.boot.context.event.ApplicationReadyEvent
+import org.springframework.boot.info.BuildProperties
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.context.event.EventListener
+
+@Configuration
+class VersionOutputter(buildProperties: BuildProperties) {
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/health/VersionOutputter.kt:12]: This is bringing inline with changes to the DPS kotlin example project.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/health/HealthInfo.kt [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import org.springframework.boot.actuate.health.Health
+import org.springframework.boot.actuate.health.HealthIndicator
+import org.springframework.boot.info.BuildProperties
+import org.springframework.stereotype.Component
+
+/**
+ * Adds version data to the /health endpoint. This is called by the UI to display API details
+ */
+@Component
+class HealthInfo(buildProperties: BuildProperties) : HealthIndicator {
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/health/HealthInfo.kt:12]: This is bringing inline with changes to the DPS kotlin example project.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/service/ImportService.kt [hunk 3]
```diff
@@ -61,6 +62,10 @@ class ImportService(
             movesTo: LocalDate,
             reportsTo: LocalDate): File? {
 
+        logger.info("Generating spreadsheet for supplier '$supplierName', moves from '$movesFrom', moves to '$movesTo' and reports to '$reportsTo'.")
+
+        if (movesFrom.plusMonths(1).isBefore(movesTo)) throw IllegalArgumentException("A maximum of one months data can be queried at a time.")
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/service/ImportService.kt:67]: Crude but hopefully enough for now.

File: helm_deploy/calculate-journey-variable-payments/templates/deployment.yaml [hunk 4]
```diff
@@ -46,15 +46,15 @@ spec:
               protocol: TCP
           livenessProbe:
             httpGet:
-              path: /actuator/health/liveness
+              path: /health/liveness
```
Comment (matt-crowson-moj) [inline helm_deploy/calculate-journey-variable-payments/templates/deployment.yaml:49]: @mafonso fyi

File: helm_deploy/calculate-journey-variable-payments/templates/deployment.yaml [hunk 5]
```diff
@@ -46,15 +46,15 @@ spec:
...
               port: {{ .Values.image.port }}
             periodSeconds: 30
             initialDelaySeconds: 90
             timeoutSeconds: 20
             failureThreshold: 10
           readinessProbe:
             httpGet:
-              path: /actuator/health/readiness
+              path: /health/readiness
```
Comment (matt-crowson-moj) [inline helm_deploy/calculate-journey-variable-payments/templates/deployment.yaml:57]: @mafonso fyi

File: src/main/resources/application.yml [hunk 6]
```diff
@@ -0,0 +1,35 @@
+spring:
```
Comment (matt-crowson-moj) [inline src/main/resources/application.yml:1]: we can review this later if need be. an intitial attempt anyway.

**Review by mafonso — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/service/ImportService.kt [hunk 7]
```diff
@@ -61,6 +62,10 @@ class ImportService(
             movesTo: LocalDate,
             reportsTo: LocalDate): File? {
 
+        logger.info("Generating spreadsheet for supplier '$supplierName', moves from '$movesFrom', moves to '$movesTo' and reports to '$reportsTo'.")
+
+        if (movesFrom.plusMonths(2).isBefore(movesTo)) throw IllegalArgumentException("A maximum of two months data can be queried at a time.")
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/service/ImportService.kt:67]: @jcavell do we defo want this.  Bumped up to two months after the workshop call today.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/output/PriceSheet.kt [hunk 8]
```diff
@@ -10,6 +10,10 @@ import java.util.concurrent.atomic.AtomicInteger
 
 abstract class PriceSheet(val sheet: Sheet, private val header: Header) {
 
+    private val formatPound = sheet.workbook.createDataFormat().getFormat("\"£\"#,##0.00_);[Red](\"£\"#,##0.00)")
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/output/PriceSheet.kt:13]: should only need to create these formats once and then reuse.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/controller/ImportControllerTest.kt [hunk 9]
```diff
@@ -50,12 +51,26 @@ class ImportControllerTest(@Autowired val restTemplate: TestRestTemplate) {
...
 
         verify(spreadsheetProtection).protectAndGet(any())
         assertThat(response.headers.contentDisposition.filename).isEqualTo("Journey_Variable_Payment_Output_SERCO_2020-10-13_15_25.xlsx")
         assertThat(response.statusCode).isEqualTo(HttpStatus.OK)
         assertThat(locationRepository.count()).isEqualTo(2)
         assertThat(priceRepository.count()).isEqualTo(1)
     }
+
+    @Test
+    fun `fails if more than one months data is requested`() {
```
Comment (jcavell) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/controller/ImportControllerTest.kt:64]: Name of test is `fails if more than one months data is requested` but it's for more than 2 months

**Review by jcavell — APPROVED** [review]

File: src/test/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/controller/ImportControllerTest.kt (hunk 9 above)
Comment (matt-crowson-moj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/controller/ImportControllerTest.kt:64]: Noted.  Will fix in future PR.

**Pull request merged**